### ISSUE TITLE
`discouraged_optional_collection` false positive in property observer

### ISSUE THREAD

**Issue by arielelkin** [realm/SwiftLint#2820]
### New Issue Checklist

- [x ] Updated SwiftLint to the latest version
- [ x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

This triggers a `discouraged_optional_collection` violation:

```swift
class MyClass {
    var items: [Int]? { 
        didSet {
            print("Hello")
        }
    }
}
```

It shouldn't in cases where there are a property observers.

**Comment by marcelofabri** [comment]
Why having a property observer should change the rule behavior?

**Comment by arielelkin** [comment]
on second thoughts i can't see why the rule's behaviour should change in that case. I think the rule itself was odd because it can be useful for a collection to be `nil`. This rule makes sense if you replace nil checks with `isEmpty` checks. Would it be useful to send a PR updating rules.md so that it adds a line doing an `isEmpty` check?

**Comment by marcelofabri** [comment]
The problem with `nil` collections is that you have two empty states: `nil` and `.isEmpty`. `Rules.md` is autogenerated, so you would need to update the respective Rule file.

**Comment by marcelofabri** [comment]
Closing this, but feel free to submit a PR improving the docs!
